File: samples/format/needs_width_100.py
"""THE ABLATION FOR ruff's `line-length = 100`.

Formatted correctly under configs/python/ruff.toml and incorrectly under ruff's
own default of 88, with line length the only setting separating the two
verdicts.

The return statement below is 95 characters wide: one line at our 100, split
across several at ruff's default 88. `bad_format.py` cannot prove this on its
own — ruff's defaults reject that file too, so it stays red even if
configs/python/ruff.toml is deleted.

Keep the long line between 89 and 100 characters or the ablation stops
separating anything.
"""


def describe_endpoint(name: str, port: int, secure: bool) -> str:
    return format_endpoint_description(name, port, secure, "the width is the entire point")


def format_endpoint_description(name: str, port: int, secure: bool, note: str) -> str:
    scheme = "https" if secure else "http"
    return f"{name}:{port} {scheme} — {note}"

File: samples/format/needs-width-100.ts
// THE ABLATION FOR `printWidth: 100`.
//
// This file is Prettier-clean under configs/typescript/prettier.config.mjs and
// Prettier-DIRTY under Prettier's own defaults, and the only setting that
// separates the two verdicts is printWidth.
//
// The call below is 96 characters wide. At our 100 it stays on one line; at
// Prettier's default 80 it is broken across four. So a CI step that runs this
// file twice — once with our config, once with none — and requires the two
// answers to differ proves the setting is load-bearing, which `bad-format.ts`
// on its own cannot do: Prettier's defaults reject that one too.
//
// If you edit this file, keep the long line between 81 and 100 characters or
// the ablation stops separating anything.

export function describe(name: string, port: number, secure: boolean): string {
  return formatEndpointDescription(name, port, secure, 'the ablation depends on this width');
}

function formatEndpointDescription(
  name: string,
  port: number,
  secure: boolean,
  note: string,
): string {
  return `${name}:${String(port)} ${secure ? 'https' : 'http'} — ${note}`;
}
